### ISSUE TITLE
fasta: Argument mismatch in super call is not correctly emitted as `InvalidInitializer`

### ISSUE THREAD

**Issue by mkustermann** [dart-lang/sdk#28301]
This code
```
class A {
  A();
  A.test(a, {b});
}

class B extends A {
  B() : super.test(b: 1);
}

main() {
  new B();
}
```

Results in:
```
  class A extends core::Object {
    constructor •() → void
      : super core::Object::•();
    constructor test(dynamic a, {dynamic b}) → void
      : super core::Object::•() ;
  }
  class B extends test::A {
    constructor •() → void
      : super test::A::test(b: 1) ;  // WRONG: Should be [InvalidInitializer]
  }
  static method main() → dynamic {
    new test::B::•();
  }
```

The `super.test(b: 1)` call is missing one required argument and should be converted to an `InvalidInitializer`.

**Comment by mkustermann** [comment]
This is form `tests/language/not_enough_positional_arguments_test.dart`

**Comment by mraleph** [comment]
I have checked and this is still the case for fasta. Assigning to @peter-ahe-google

**Comment by jensjoha** [comment]
The current result is
```
class A extends core::Object {
  constructor •() → void
    : super core::Object::•()
    ;
  constructor test(dynamic a, {dynamic b = null}) → void
    : super core::Object::•()
    ;
}
class B extends self::A {
  constructor •() → void
    : final dynamic #t1 = throw new core::NoSuchMethodError::_withType(null, #test, 0, <dynamic>[].toList(growable: false), <dynamic, dynamic>{#b: 1}, null)
    ;
}
static method main() → dynamic {
  new self::B::•();
}
```

which seems to fit with the [specification](http://www.ecma-international.org/publications/files/ECMA-ST/ECMA-408.pdf)
```
16.17.3 Super Invocation
[...]
If n < h, or n > m, the method lookup has failed.
[...]
If getter lookup has also failed, then a new instance im of the predefined class Invocation is created [...] Then the method noSuchMethod() is looked up in Sdynamic and invoked on this
with argument im [...]
```

**Comment by peter-ahe-google** [comment]
This was fixed in e2eee219acd5b19459bbe4cf266c2ad50b85566e.
